TFUtil.py

```diff
@@ -937,10 +953,7 @@ def get_placeholder_flattened(self, keep_dims=False):
                   for i in dyn_axes]
       ndim -= 1
     if len(dyn_axes) > 1:
-      assert 0 in dyn_axes, "would need some transpose, not supported at the moment"
-      for i in dyn_axes:
-        if i > 0:
-          assert i - 1 in dyn_axes, "would need some transpose, not supported at the moment"
+      # Transpose x to get dyn axes first.
```

Comment:
I'm pretty sure now that it will not be correct with `0 not in dyn_axes`, because `flatten_with_seq_len_mask` will always have the new resulting flattened axis in 0.
Also, in your use case, you will anyway have `0 in dyn_axes`. So just leave the check in.

Reply:
Also add a test case for `get_placeholder_flattened`.

Reply:
Well, I think the code will work if `0 not in dyn_axes` but I didn't test that. And I doubt anybody needs this case. That's why there's  `assert 0 in [self.time_dim_axis, self.batch_dim_axis]` in the `get_placeholder_time_flattened()`.